src/components/pacbio/PacbioLibraryCreate.vue

```diff
@@ -107,80 +107,97 @@
   </div>
 </template>
 
-<script>
-import ModalHelper from '@/mixins/ModalHelper'
-import { createNamespacedHelpers } from 'vuex'
-const { mapActions, mapGetters } = createNamespacedHelpers('traction/pacbio/libraries')
-
-export default {
-  name: 'PacbioLibraryCreate',
-  mixins: [ModalHelper],
-  props: {
-    disabled: Boolean,
-    isStatic: Boolean,
-    selectedSample: {
-      type: Object,
-      default() {
-        return {}
-      },
+<script setup>
+import { computed, ref } from 'vue'
+import { usePacbioLibrariesStore } from '@/stores/pacbioLibraries'
+import useAlert from '@/composables/useAlert.js'
+import useModalHelper from '@/composables/useModalHelper.js'
+
+// Define props
+const props = defineProps({
+  disabled: Boolean,
+  isStatic: Boolean,
+  selectedSample: {
+    type: Object,
+    default() {
+      return {}
     },
   },
-  emits: ['alert'],
-  data() {
-    return {
-      library: { tag: { id: '' }, sample: {} },
-      selectedTagSetId: '',
-      showModal: false,
+})
+
+// Define refs
+const library = ref({ tag: { id: '' }, sample: {} })
+const selectedTagSetId = ref('')
+const showModal = ref(false)
+const modalRef = ref(null)
+
+//Composables
+const { showAlert } = useAlert()
+const { hide } = useModalHelper(modalRef.value)
+
+// Define emits
+const emit = defineEmits(['alert'])
+//Emits the 'alert' event
+const emitAlert = (message) => {
```

Comment:
This emit also needs to emit the alert type. Currently if you create a library you get a default gray alert colour but it should be green ('success'). I believe this is where it is getting blocked from.

Reply:
You can probably remove this emitAlert method and call emit('alert') directly where its used as it is only used once.